From 77e789ae04d05a0279d16c6aea47754c330727fc Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:29 -0400
Subject: [ethernet] bnx2x: Fix failure to configure VF multicast filters

Message-id: <1409946615-5862-25-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91560
O-Subject: [RHEL7.1 PATCH 24/70] bnx2x: Fix failure to configure VF multicast filters
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit ab15f86b8d3c3e7a1437010bb57ea543fc422463
Author: Narender Kumar <narender.kumar@qlogic.com>
Date:   Thu Apr 24 19:29:54 2014 +0300

    bnx2x: Fix failure to configure VF multicast filters

    Commit 2dc33bbc "bnx2x: Remove the sriov VFOP mechanism" caused a regression,
    preventing VFs from configuring multicast filters.

    Signed-off-by: Naredner Kumar <narender.kumar@qlogic.com>
    Signed-off-by: Yuval Mintz <Yuval.Mintz@qlogic.com>
    Signed-off-by: Ariel Elior <Ariel.Elior@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c
index 9ba3371..addb79c 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.c
@@ -610,6 +610,7 @@ int bnx2x_vf_mcast(struct bnx2x *bp, struct bnx2x_virtf *vf,
   }
 
   /* add new mcasts */
+  mcast.mcast_list_len = mc_num;
   rc = bnx2x_config_mcast(bp, &mcast, BNX2X_MCAST_CMD_ADD);
   if (rc)
    BNX2X_ERR("Faled to add multicasts\n");
-- 
1.7.1